FWCore/Framework/interface/EDConsumerBase.h

```diff
@@ -97,7 +99,9 @@ namespace edm {
     typedef ProductLabels Labels;
     void labelsForToken(EDGetToken iToken, Labels& oLabels) const;
 
-    void modulesWhoseProductsAreConsumed(std::vector<ModuleDescription const*>& modules,
+    void modulesWhoseProductsAreConsumed(std::vector<ModuleDescription const*>& modulesEvent,
+                                         std::vector<ModuleDescription const*>& modulesLumiRun,
+                                         std::set<ModuleProcessName>& modulesInPreviousProcesses,
```

Comment:
Does this gather info from both parent Processes (assuming the module itself is in a SubProcess) as well as dependencies on data in the Source? Or is this only parents?

Reply:
It contains also dependencies on data in the Source, basically the (module label, process name) pairs for all consumed products that either can not be satisfied by the producers in the SubProcess (no such module label or process name is different), or may be satisfied by the producers in "earlier processes" (process name is empty or instructed to skip the current process).